packages/components/forms/src/text-input/TextInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, ChangeEvent } from 'react';
+import { BaseInput } from '../base-input';
+import { Box, Flex } from '@contentful/f36-core';
+import { HelpText } from '@contentful/f36-helptext';
+import { CopyButton } from '@contentful/f36-copybutton';
+import getStyles from './TextInput.styles';
+import { TextInputProps } from './types';
+import { cx } from 'emotion';
+
+export const _TextInput = (
+  {
+    className,
+    testId = 'cf-ui-text-input',
+    countCharacters = false,
+    label,
+    id,
+    value,
+    onChange,
+    maxLength,
+    isInvalid,
+    isDisabled,
+    withCopyButton,
+    onCopy,
+    ...otherProps
+  }: TextInputProps,
+  ref: React.Ref<HTMLInputElement>,
+) => {
+  const [valueState, setValueState] = useState<string | undefined>(value);
+  const styles = getStyles();
+
+  // Store a copy of the value in state.
+  // This is used by this component when the `countCharacters`
+  // option is on
+  const handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setValueState(e.target.value);
+    if (onChange) onChange(e);
+  };
+
+  const copyButtonStyles = cx(styles.copyButton, {
+    [styles.disabled]: Boolean(isDisabled),
+    [styles.invalid]: Boolean(isInvalid),
+  });
+
+  const inputWithCopyButton = (
+    <Flex className={className}>
+      <BaseInput
+        {...otherProps}
+        testId={testId}
+        ref={ref}
+        label={label}
+        type="text"
+        onChange={handleOnChange}
+        as="input"
+        maxLength={maxLength}
+        className={styles.inputWithCopyButton}
+      />
+      <CopyButton
+        value={valueState}
+        onCopy={onCopy}
+        className={copyButtonStyles}
+      />
+    </Flex>
+  );
+
+  return (
+    <Flex flexDirection="column" fullWidth className={className}>
+      {withCopyButton ? (
+        inputWithCopyButton
+      ) : (
+        <BaseInput
+          testId={testId}
+          ref={ref}
+          label={label}
+          type="text"
+          onChange={handleOnChange}
+          as="input"
+          isInvalid={isInvalid}
+          isDisabled={isDisabled}
+          maxLength={maxLength}
+          {...otherProps}
+        />
+      )}
+
+      {countCharacters && maxLength && (
+        <Flex justifyContent="flex-end">
+          <Box marginTop="spacingXs" marginLeft="spacingM">
+            <HelpText>
+              {valueState ? valueState.length : 0}/{maxLength}
+            </HelpText>
+          </Box>
+        </Flex>
+      )}
```

Comment:
Uhmmmmmm 🤔 

I'm thinking now that we should do something else here

Because of this case:
![Screenshot 2021-08-31 at 16 25 41](https://user-images.githubusercontent.com/6597467/131520645-8028fdc9-7015-44f7-bf8e-e69aa7660b73.png)

the counter needs to be aligned with ValidationMessage

Maybe we should build TextInput without the counter for now, and in another PR build an example with `FormControl` + `ValidationMessage` and then implement the counter somehow